osu.Game.Rulesets.Osu.Tests/Editor/TestSceneSliderSnapping.cs

```diff
@@ -61,9 +63,14 @@ public override void SetUpSteps()
             }));
             AddStep("set beat divisor to 1/1", () =>
             {
-                var beatDivisor = (BindableBeatDivisor)Editor.Dependencies.Get(typeof(BindableBeatDivisor));
+                var beatDivisor = Editor.Dependencies.Get<BindableBeatDivisor>();
                 beatDivisor.Value = 1;
             });
+            AddStep("set distance spacing to 1", () =>
+            {
+                var distanceSpacing = (BindableDouble)Editor.ChildrenOfType<OsuHitObjectComposer>().Single().DistanceSpacingMultiplier;
+                distanceSpacing.Value = 1;
+            });
```

Comment:
While looking at each case of this, I've realized this is actually hiding a change in behaviour between osu!(stable) and osu!(lazer):

In osu!(stable), the distance spacing multipler does not affect the snapping of the slider:

https://user-images.githubusercontent.com/22781491/165670957-d6eed6e1-ebff-4cf2-afb8-16d9ce1d825b.mp4

Meanwhile on osu!(lazer) with this change, it does affect the slider snapping:

https://user-images.githubusercontent.com/22781491/165670990-cfb81127-768b-49cd-b02a-b61daf08c4f2.mp4

I'm not entirely sure which makes more sense here, I would think osu!(lazer) is better since it's making snap distances consistent across hitcircles and sliders? maybe, or maybe that doesn't make sense (cc @peppy).

Reply:
Distance snap should only be applied to the head of a slider (ie. before the slider path placement begins). In your video, it looks like lazer is incorrectly attempting to distance snap the tail to the head or something? Shouldn't be a thing.

Slider placement is only affected by the beat snap divisor and slider velocity. Distance snap is specifically for the distance between adjacent hitobjects. To explain another way, with distance snap set to 1.0x, objects should be spaced apart at the same rate as a slider's ball moves down the path.